Adafruit_CAP1188.cpp

```diff
@@ -1,48 +1,74 @@
-/*************************************************** 
-  This is a library for the CAP1188 I2C/SPI 8-chan Capacitive Sensor
-
-  Designed specifically to work with the CAP1188 sensor from Adafruit
-  ----> https://www.adafruit.com/products/1602
-
-  These sensors use I2C/SPI to communicate, 2+ pins are required to  
-  interface
-  Adafruit invests time and resources providing this open source code, 
-  please support Adafruit and open-source hardware by purchasing 
-  products from Adafruit!
-
-  Written by Limor Fried/Ladyada for Adafruit Industries.  
-  BSD license, all text above must be included in any redistribution
- ****************************************************/
+/*!
+ *  @file Adafruit_CAP1188.cpp
+ *
+ *  @mainpage Adafruit CAP1188 I2C/SPI 8-chan Capacitive Sensor
+ *
+ *  @section intro_sec Introduction
+ *
+ * 	This is a library for the Adafruit CAP1188 I2C/SPI 8-chan Capacitive
+ * Sensor http://www.adafruit.com/products/1602
+ *
+ *  These sensors use I2C/SPI to communicate, 2+ pins are required to
+ *  interface
+ *
+ * 	Adafruit invests time and resources providing this open source code,
+ *  please support Adafruit and open-source hardware by purchasing products from
+ * 	Adafruit!
+ *
+ *  @section author Author
+ *
+ *  Limor Fried/Ladyada (Adafruit Industries).
+ *
+ * 	@section license License
+ *
+ * 	BSD (see license.txt)
+ *
+ * 	@section  HISTORY
+ *
+ *     v1.0 - First release
+ */
 
 #include "Adafruit_CAP1188.h"
+#include <Wire.h>
 
 // If the SPI library has transaction support, these functions
 // establish settings and protect from interference from other
 // libraries.  Otherwise, they simply do nothing.
 #ifdef SPI_HAS_TRANSACTION
-  static inline void spi_begin(void) __attribute__((always_inline));
-  static inline void spi_begin(void) {
-    // max speed!
-    SPI.beginTransaction(SPISettings(2000000, MSBFIRST, SPI_MODE0));
-  }
-  static inline void spi_end(void) __attribute__((always_inline));
-  static inline void spi_end(void) {
-    SPI.endTransaction();
-  }
-#else
-  #define spi_begin()
-  #define spi_end()
+static inline void spi_begin(void) __attribute__((always_inline));
+static inline void spi_begin(void) {
+  // max speed!
+  SPI.beginTransaction(SPISettings(2000000, MSBFIRST, SPI_MODE0));
+}
+static inline void spi_end(void) __attribute__((always_inline));
+static inline void spi_end(void) { SPI.endTransaction(); }
+/* #else */
+/*   #define spi_begin() */
+/*   #define spi_end() */
 #endif
 
+/* uint8_t mySPCR, SPCRback; */
 
-uint8_t mySPCR, SPCRback;
-
+/*!
+ *    @brief  Instantiates a new CAP1188 class using hardware I2C
+ *    @param  resetpin
+ *            number of pin where reset is connected
+ *
+ */
 Adafruit_CAP1188::Adafruit_CAP1188(int8_t resetpin) {
   // I2C
   _resetpin = resetpin;
   _i2c = true;
 }
 
+/*!
```

Comment:
add optional SPI pointer device (default to &SPI) so they can pass in optional SPI device

Reply:
do we have any library that have optional spi pointer param? would be cool if we clean up this to be consistent.

Reply:
OK, please look is it done right :).
Also, imho we can remove all `SPI_HAS_TRANSACTION` from the code and add one which just throws error if `!SPI_HAS_TRANSACTION` but looking forward for your approve :)